content/program-security.md

```diff
@@ -0,0 +1,724 @@
+
+For example, imagine that the character referenced in these examples is an NFT. Further, imagine that the program allows the NFT to be staked to earn token rewards proportional to the NFTs number of attribute points. Failure to implement these data validation checks would allow a bad actor to assign an obscenely high number of attribute points and quickly drain your treasury of all the rewards that were meant to be spread more evenly amongst a larger pool of stakers.
+
+### Integer overflow and underflow
+
+Rust integers have fixed sizes. This means they can only support a specific range of numbers. An arithmetic operation that results in a higher or lower value than what is supported by the range will cause the resulting value to wrap around. For example, a `u8` only supports numbers 0-255, so the result of addition that would be 256 would actually be 0, 257 would be 1, etc.
+
+This is always important to keep in mind, but especially so when dealing with any code that represents true value, such as depositing and withdrawing tokens.
+
+To avoid integer overflow and underflow, either:
+
+1. Have logic in place that ensures overflow or underflow *cannot* happen or
+2. Use checked math like `checked_add` instead of `+`
+	```rust
+    let first_int: u8 = 5;
+    let second_int: u8 = 255;
+    let sum = first_int.checked_add(second_int);
+	```
+
+# Demo
+
+Let’s practice together with the Movie Review program we've worked on in previous lessons. No worries if you’re just jumping into this lesson without having done the previous lesson - it should be possible to follow along either way.
+
+As a refresher, the Movie Review program lets users store movie reviews in PDA accounts. Last lesson, we finished implementing the basic functionality of adding a movie review. Now, we'll add some security checks to the functionality we've already created and add the ability to update a movie review in a secure manner.
+
+Just as before, we'll be using [Solana Playground](https://beta.solpg.io/) to write, build, and deploy our code.
+
+## 1. Get the starter code
+
+To begin, you can find the starter code [here](https://beta.solpg.io/62b552f3f6273245aca4f5c9). If you've been following along with the Movie Review demos, you'll notice that we've refactored our program.
+
+The refactored starter code is almost the same as what it was before. Since `lib.rs` was getting rather large and unwieldy, we've separated its code into 3 files: `lib.rs`, `entrypoint.rs`, and `processor.rs`. `lib.rs` now *only* registers the code's modules, `entrypoint.rs` *only* defines and sets the program's entrypoint, and `processor.rs` handles the program logic for processing instructions. We've also added an `error.rs` file where we'll be defining custom errors. The complete file structure is as follows:
+
+- **lib.rs** - register modules
```

Comment:
@jamesrp13 alphabetical order? e.g.

- **entrypoint.rs -** entry point to the program
- **error.rs -** custom program errors
- **instruction.rs -** serialize and deserialize instruction data
- **lib.rs** - register modules
- **processor.rs -** program logic to process instructions
- **state.rs -** serialize and deserialize state